code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -269,16 +269,24 @@
 			return
 	//SKYRAT EDIT ADDITION BEGIN - EMOTES
 	if(zone_selected == BODY_ZONE_PRECISE_GROIN && target.dir == src.dir)
-		if(HAS_TRAIT(target, TRAIT_IRONASS))
-			var/obj/item/bodypart/affecting = src.get_bodypart("[(src.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
-			if(affecting?.receive_damage(2))
-				src.update_damage_overlays()
-			visible_message("<span class='danger'>[src] tried slapping [target]'s ass, however it was much harder than expected!</span>",
-			"<span class='danger'>You tried slapping [target]'s ass, but it felt like metal, ouch!</span>",\
-			"You hear a sore sounding slap.", ignored_mobs = list(target))
-			playsound(target.loc, 'sound/effects/snap.ogg', 50, TRUE, -1)
-			to_chat(target, "<span class='danger'>[src] tried slapping your ass, but it was deflected!")
-			return
+		if(HAS_TRAIT(target, TRAIT_PERSONALSPACE) && (target.stat != UNCONSCIOUS) && (!target.handcuffed)) //You need to be conscious and uncuffed to use Personal Space
+			if(target.combat_mode && (!HAS_TRAIT(target, TRAIT_PACIFISM))) //Being pacified prevents violent counters
+				var/obj/item/bodypart/affecting = src.get_bodypart(BODY_ZONE_HEAD)
+				if(affecting?.receive_damage(2))
+					src.update_damage_overlays()
+				visible_message(span_danger("[src] tried slapping [target]'s ass, but they were slapped instead!"),
+				span_danger("You tried slapping [target]'s ass, but they hit you back, ouch!"),
+				"You hear a slap.", ignored_mobs = list(target))
+				playsound(target.loc, 'sound/effects/snap.ogg', 50, TRUE, -1)
```

Comment:
What does the `-1` mean here? Surely there's a define for it somewhere.

Reply:
-1 is the extrarange var of the proc. Should this also be a constant?

Reply:
Hmmmm... I mean, ideally, yes, I think so, but I wouldn't be mad if it wasn't.

Reply:
```suggestion
				playsound(target.loc, 'sound/effects/snap.ogg', 50, TRUE, ASS_SLAP_EXTRA_RANGE)
```